src/app/clusters/occupancy-sensor-server/README.md

```diff
@@ -0,0 +1,157 @@
+
+For new applications using the `CodeDrivenDataModelProvider`, we strongly
+recommend instantiating and registering the cluster directly. This approach
+provides the most flexibility and control.
+
+### 1. Implement the Delegate (Optional)
+
+If your application needs to be notified of changes, create a class that
+inherits from `chip::app::Clusters::OccupancySensingDelegate` and implement its
+virtual methods.
+
+```cpp
+#include "app/clusters/occupancy-sensor-server/OccupancySensingCluster.h"
+
+class MyOccupancyDelegate : public chip::app::Clusters::OccupancySensingDelegate
+{
+public:
+    void OnOccupancyChanged(bool occupied) override
+    {
+        // Your logic to react to an occupancy change
+    }
+
+    void OnHoldTimeChanged(uint16_t holdTime) override
+    {
+        // Your logic to react to a hold time change
+    }
+};
+```
+
+### 2. Instantiate Delegates and Cluster
+
+Instantiate your optional delegate, a timer delegate (if using the Hold Time
+feature), and the `OccupancySensingCluster` itself for each required endpoint.
+Using `RegisteredServerCluster` simplifies registration.
+
+The `HoldTime` feature and its related attributes are optional. To enable them,
+call the `.WithHoldTime()` method on the `Config` object. If it is not called,
+these attributes will be disabled.
+
+```cpp
+#include "app/DefaultTimerDelegate.h"
+#include "app/clusters/occupancy-sensor-server/OccupancySensingCluster.h"
+#include "app/server-cluster/ServerClusterInterfaceRegistry.h"
+
+// In a .cpp file
+MyOccupancyDelegate gMyOccupancyDelegate;
+DefaultTimerDelegate gTimerDelegate;
+
+constexpr chip::app::Clusters::OccupancySensing::Structs::HoldTimeLimitsStruct::Type kHoldTimeLimits = {
+    .holdTimeMin = 1, .holdTimeMax = 300, .holdTimeDefault = 60
+};
+
+chip::app::RegisteredServerCluster<chip::app::Clusters::OccupancySensingCluster> gOccupancyCluster(
+    chip::app::Clusters::OccupancySensingCluster::Config(kYourEndpointId)
+        .WithFeatures(chip::app::Clusters::OccupancySensing::Feature::kPassiveInfrared)
+        .WithHoldTime(60, kHoldTimeLimits, gTimerDelegate) // Enable HoldTime feature
+        .WithDelegate(gMyOccupancyDelegate)                // Attach optional delegate
+);
+```
+
+### 3. Register the Cluster
+
+In your application's initialization sequence, register the cluster instance
+with either the `CodegenDataModelProvider` (legacy) or the
+`CodeDrivenDataModelProvider`.
+
+```cpp
+#include "data-model-providers/codegen/CodegenDataModelProvider.h"
+
+void ApplicationInit()
+{
+    // ... other initializations
+    CHIP_ERROR err = chip::app::CodegenDataModelProvider::Instance().Registry().Register(gOccupancyCluster.Registration());
+    // ...
+}
+```
+
+### 4. Notify the Cluster of Occupancy Changes
+
+Your application is responsible for monitoring the physical sensor. When the
+sensor state changes, call the `SetOccupancy()` method on the cluster instance.
+
+```cpp
+void MySensorHardwareCallback(bool isOccupied)
```

Comment:
Also need an example of when it's based on legacy Ember callbacks and global ::Set calls on the occupancy attribute. Some guide of how to take a previous sample and making it work would be great to have, that also properly gets the instance from the endpoint using calls like your new get cluster on endpoint.

You have to allude that when transitioning to the new method, you should not maintain the holdtime timer on your application directly,